### PR TITLE
Support for StringMap and Duration attribute types

### PR DESCRIPTION
**Pull request by qiwzhang**

### CODE REVIEW

File: include/client.h [hunk 1]
```diff
@@ -66,7 +66,15 @@ struct Attributes {
   // A structure to hold different types of value.
   struct Value {
     // Data type
-    enum ValueType { STRING, INT64, DOUBLE, BOOL, TIME, BYTES } type;
+    enum ValueType {
+      STRING,
+      INT64,
+      DOUBLE,
+      BOOL,
+      TIME,
```
Comment (geeknoid) [inline include/client.h]: NB: we also need support for DURATION.

File: src/signature.cc [hunk 2]
```diff
@@ -56,6 +56,9 @@ string GenerateSignature(const Attributes& attributes) {
         hasher.Update(&attribute.second.time_v,
                       sizeof(attribute.second.time_v));
         break;
+      // Not to calculate signature based on STRING_MAP type.
```
Comment (geeknoid) [inline src/signature.cc]: Why isn't the string map included?

File: include/client.h (hunk 1 above)
Comment (geeknoid) [inline include/client.h]: NB: we need to also add support for DURATION.

Comment (qiwzhang) [inline include/client.h]: Done

Comment (qiwzhang) [inline include/client.h]: Done

File: src/signature.cc (hunk 2 above)
Comment (qiwzhang) [inline src/signature.cc]: Added.

File: include/client.h [hunk 3]
```diff
@@ -86,6 +87,7 @@ struct Attributes {
     // It is not easy for union to support them.
     std::string str_v;  // for both STRING and BYTES
     std::chrono::time_point<std::chrono::system_clock> time_v;
+    ::google::protobuf::Duration duration;
```
Comment (geeknoid) [inline include/client.h]: Shouldn't this be std::chrono::duration, to be consistent with how time is handled? Alternatively, the time value could be changed to ::google::protobuf::Timestamp. Either way works, but I do think we should be consistent.

Comment (geeknoid) [inline include/client.h]: Also, I suggest updating the PR description to mention DURATION.

Comment (qiwzhang) [inline include/client.h]: Done

**Pull request merged**